include/zephyr/bluetooth/mesh/main.h

```diff
@@ -173,6 +173,7 @@ struct bt_mesh_prov {
 	 */
 	void         (*capabilities)(const struct bt_mesh_dev_capabilities *cap);
 
+#if defined CONFIG_BT_MESH_REDUCED_MAX_OOB_SIZE
 	/** @brief Output of a number is requested.
```

Comment:
We need to add here that this callback is enabled only if `CONFIG_BT_MESH_REDUCED_MAX_OOB_SIZE` is enabled.